homeassistant/components/shelly/climate.py

```diff
@@ -0,0 +1,201 @@
+        return cast(float, self.block.temp)
+
+    @property
+    def available(self) -> bool:
+        """Device availability."""
+        return not cast(bool, self._device_block.valveError)
+
+    @property
+    def hvac_mode(self) -> str:
+        """HVAC current mode."""
+        if self._device_block.mode is None or self._check_is_off():
+            return HVAC_MODE_OFF
+
+        return HVAC_MODE_HEAT
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Preset current mode."""
+        if self._device_block.mode is None:
+            return None
+        return self._attr_preset_modes[cast(int, self._device_block.mode)]
+
+    @property
+    def hvac_action(self) -> str | None:
+        """HVAC current action."""
+        if self._device_block.status is None or self._check_is_off():
+            return CURRENT_HVAC_OFF
+
+        return (
+            CURRENT_HVAC_IDLE if self._device_block.status == 0 else CURRENT_HVAC_HEAT
+        )
+
+    def _check_is_off(self) -> bool:
+        """Return if valve is off or on."""
+        return bool(
+            self.target_temperature is None
+            or (self.target_temperature <= self._attr_min_temp)
+        )
+
+    def _load_device_block(self) -> bool:
+        """Load device CoAP block."""
+        assert self.wrapper.device.blocks
+
+        for dev_block in self.wrapper.device.blocks:
+            if dev_block.type == "device":
+                self._device_block = dev_block
+
+        assert self._device_block
+        return True
+
+    async def async_update(self) -> None:
+        """Update entity with latest info."""
+        await self.wrapper.async_request_refresh()
+        await self.hass.async_add_executor_job(self._load_device_block)
+
+    async def set_state_full_path(self, **kwargs: Any) -> Any:
+        """Set block state (HTTP request)."""
+        _LOGGER.debug("Setting state for entity %s, state: %s", self.name, kwargs)
+        try:
+            async with async_timeout.timeout(AIOSHELLY_DEVICE_TIMEOUT_SEC):
+                return await self.wrapper.device.http_request(
+                    "get", f"thermostat/{self.block.channel}", kwargs
+                )
+        except (asyncio.TimeoutError, OSError) as err:
+            _LOGGER.error(
+                "Setting state for entity %s failed, state: %s, error: %s",
+                self.name,
+                kwargs,
+                repr(err),
+            )
+            self.wrapper.last_update_success = False
+            return None
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        if (current_temp := kwargs.get(ATTR_TEMPERATURE)) is None:
+            return
+        await self.set_state_full_path(target_t_enabled=1, target_t=f"{current_temp}")
+        if self._check_is_off():
+            self.async_set_hvac_mode(HVAC_MODE_OFF)
+        else:
+            self.async_set_hvac_mode(HVAC_MODE_HEAT)
```

Comment:
Generally we don't expect the external library to use Home Assistant's internal constants.

We should import the constants from the library and map the Home Assistant constant to the shelly constant.

I realize the library is probably built for Home Assistant only so they may be the same, but there is some chance we change them later in Home Assistant and shelly will get unexpected values.

Reply:
Discussing on discord...